ui/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -183,30 +194,65 @@ export default class PermissionConnect extends Component {
       () => this.props.history.push(confirmPermissionPath),
       ///: END:ONLY_INCLUDE_IN
       ///: BEGIN:ONLY_INCLUDE_IN(flask)
-      () =>
-        this.props.history.push(
-          // eslint-disable-next-line no-nested-ternary
-          isSnap
-            ? permissionsRequest.newPermissions
-              ? snapUpdatePath
-              : snapInstallPath
-            : confirmPermissionPath,
-        ),
+      () => {
+        switch (requestType) {
+          case 'wallet_installSnap':
+            this.props.history.push(snapInstallPath);
+            break;
+          case 'wallet_updateSnap':
+            this.props.history.push(snapUpdatePath);
+            break;
+          case 'wallet_installSnapResult':
+            this.props.history.push(snapResultPath);
+            break;
+          default:
+            this.props.history.push(confirmPermissionPath);
```

Comment:
nit: you can break this off into a separate function e.g. `routeRequest` since you're using the same logic above